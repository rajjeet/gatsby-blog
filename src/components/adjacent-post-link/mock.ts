export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
