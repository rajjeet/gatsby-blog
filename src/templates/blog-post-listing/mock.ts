import { mockPosts } from '../../fixtures/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: mockPosts,
    },
  },
  pageContext: {
    numOfPages: 2,
    currentPage: 3,
    tag: 'Sample Tag',
  },
});
