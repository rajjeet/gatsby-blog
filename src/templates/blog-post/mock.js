import { createMockPosts } from '../../utils/testing';

export const makeProps = () => ({
  className: {},
  data: {
    post: createMockPosts[1].node,
  },
});
