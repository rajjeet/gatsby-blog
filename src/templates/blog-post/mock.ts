import { TProps } from './types';
import { mockPost } from '../../fixtures/testing';

export const makeProps = ({ dateModified = null } = {}): TProps => {
  if (dateModified) {
    mockPost.frontmatter.dateModified = dateModified;
  }
  return {
    data: {
      post: { ...mockPost },
    },
  };
};
