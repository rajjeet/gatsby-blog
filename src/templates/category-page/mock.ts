import { createMockPosts } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    category: 'Self Improvement',
    numOfPages: 2,
    currentPage: 3,
    paginationSlug: '/self-improvement/',
  },
});
