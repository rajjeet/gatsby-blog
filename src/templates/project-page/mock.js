import { createMockPosts, createMockProjects } from '../../utils/testing';

export const makeProps = () => ({
  data: {
    project: createMockProjects[0].node,
    posts: {
      edges: createMockPosts,
    },
  },
});
