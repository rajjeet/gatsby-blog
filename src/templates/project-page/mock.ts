import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';
import { mdxBody } from '../../fixtures/mdx';

export const makeProps = (): TProps => ({
  data: {
    project: {
      frontmatter: {
        title: 'Condo Landing Pages',
        description: 'An automated process for building fast, responsive landing pages for pre-construction condos.',
        techStackTags: [
          {
            type: 'AWS',
            label: 'AWS CloudFormation',
          },
        ],
        links: [
          {
            label: 'Github Code',
            value: 'https://github.com/rajjeet/charismacondos',
          },
        ],
      },
      body: mdxBody,
    },
    posts: {
      edges: mockPosts,
    },
  },
});
