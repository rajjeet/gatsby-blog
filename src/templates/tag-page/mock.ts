import { createMockPosts } from '../../utils/testing';
import { TProps } from './index';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    limit: 5,
    skip: 2,
    numOfPages: 2,
    numOfPosts: 10,
    currentPage: 3,
    tag: 'Life Skills',
    paginationSlug: '',
  },
});
