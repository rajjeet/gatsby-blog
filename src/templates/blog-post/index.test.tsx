import React from 'react';
import { render } from '@testing-library/react';
import prismjs from 'prismjs';
import tocbot from 'tocbot';
import BlogPost from './index';
import { makeProps } from './mock';

describe('BlogPost', () => {
  it('should have a table of contents', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText('Outline')).toBeDefined();
  });

  it('should show a Share links section', () => {
    const { queryAllByText } = render(<BlogPost {...makeProps()} />);
    expect(queryAllByText('Share').length).toBeGreaterThan(0);
  });

  describe('post summary', () => {
    it('should show the post title', () => {
      const { getByText } = render(<BlogPost {...makeProps()} />);
      expect(getByText('State Management using React Hooks')).toBeDefined();
    });

    it('should show the post date created', () => {
      const { getByText } = render(<BlogPost {...makeProps()} />);
      expect(getByText(/02 May, 2019/)).toBeDefined();
    });

    it('should show the time to read', () => {
      const { getByText } = render(<BlogPost {...makeProps()} />);
      expect(getByText(/4 mins/)).toBeDefined();
    });

    it('should show the description', () => {
      const { getByText } = render(<BlogPost {...makeProps()} />);
      expect(getByText(/I was learning about the new React Hooks/)).toBeDefined();
    });

    it('should shows the blog tag', () => {
      const { getByText } = render(<BlogPost {...makeProps()} />);
      expect(getByText('React X')).toBeDefined();
      expect(getByText('Angular X')).toBeDefined();
    });

    it('should not show Updated detail when dateModified is not specified', () => {
      const { queryByText } = render(<BlogPost {...makeProps({ dateModified: null })} />);
      expect(queryByText('Updated')).toBeNull();
    });

    it('should show Updated detail when dateModified is specified', () => {
      const { queryByText } = render(<BlogPost {...makeProps({ dateModified: '2020-01-01' })} />);
      expect(queryByText('Updated')).not.toBeNull();
    });

    it('should initialize tocbot on load', () => {
      const spyInstance = jest.spyOn(tocbot, 'init');
      jest.spyOn(document, 'getElementById').mockImplementation(() => document.createElement('div'));
      render(<BlogPost {...makeProps({})} />);
      expect(spyInstance).toHaveBeenCalledTimes(1);
    });

    it('should load prism highlights on load', () => {
      const spyInstance = jest.spyOn(prismjs, 'highlightAll');
      jest.spyOn(document, 'getElementById').mockImplementation(() => document.createElement('div'));
      render(<BlogPost {...makeProps({})} />);
      expect(spyInstance).toHaveBeenCalledTimes(1);
    });

    it('should empties the static table of contents on load', () => {
      const { getByTestId } = render(<BlogPost {...makeProps({})} />);
      expect(getByTestId('static-toc').textContent).toEqual('');
    });

    it('should empty the static table of contents on load', () => {
      render(<BlogPost {...makeProps({})} />);
    });
  });
});
