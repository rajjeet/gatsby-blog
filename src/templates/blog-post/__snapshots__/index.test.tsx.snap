// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should render 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat faJHDz"
  >
    <div
      class="sc-bwzfXH bmdiOC"
    >
      <a
        aria-label="See Home"
        href="/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden; width: 3em;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Ortmesh logo"
              loading="lazy"
              sizes=""
              src="https://via.placeholder.com/250x180"
              srcset=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Ortmesh logo" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </a>
      <a
        href="/"
      >
        <h2>
          Ortmesh
        </h2>
        <p>
          Write code that matters
        </p>
      </a>
      <button
        class="sc-bdVaJa eSxrIN"
      >
        About
      </button>
    </div>
    <div
      class="sc-VigVT gNeyZm"
    >
      <div
        class="post-summary"
      >
        <h1>
          State Management using React Hooks
        </h1>
        <span>
          02 May, 2019 - 4 min read - 
          <span
            class="disqus-comment-count"
            data-disqus-identifier="5f2933ac-25ed-5ae0-84d7-c5d59fc4448e"
            data-disqus-url="http://ortmesh.com/state-management-react-hooks/"
          >
            Comments
          </span>
        </span>
        <p>
          I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.
        </p>
        <div
          class="sc-bxivhb fXNsAS"
        >
          <span
            role="link"
            tabindex="0"
          >
            React X
          </span>
          <span
            role="link"
            tabindex="0"
          >
            Angular X
          </span>
        </div>
      </div>
      <br />
      <div
        class="sc-iwsKbI OzSrb"
      >
        <div
          class="sc-gqjmRU fmitny"
        >
          <div
            class="js-toc-content"
          />
          <br />
          <div
            id="disqus_thread"
          />
        </div>
        <div
          class="sc-gZMcBi dzpiJb"
        >
          <div
            class="sc-dnqmqq gCNeSl"
          >
            <h4>
              Outline
            </h4>
            <ul
              class="sc-EHOje kngaEo"
              id="static-toc"
            />
            <div
              class="js-toc"
            />
          </div>
          <div
            class="sc-htoDjs eIKyiD"
          >
            <button
              aria-label="Open table of contents"
              class="sc-ifAKCX hxQEAp"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-list fa-w-16 fa-2x "
                color="white"
                data-icon="list"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
